### ISSUE TITLE
Spectator suggested moves

### ISSUE THREAD

**Issue by jacoblangston** [arbrown/pao#65]
Allow spectators to suggest moves by double clicking a cell. The moves should show up in chat with the name of the people that suggested it.

**Comment by walrii** [comment]
Rather than showing up in chat, they could show up as "+N" next to the piece, where N is the number of recommendations (hover shows who).  Might want an option to hide them.

OR When a kibitzer right clicks on a piece, up pops a small voting booth, with up/down arrows.

**Comment by arbrown** [comment]
I do like this idea.  I think instead of in chat, it should overlay an arrow on the board that fades out or can be dismissed with a click.  It can't just be a double click though, since some moves involve one square (turn up) and others (usually the most interesting) involve two.

I think this might be pretty straightforward to implement though.  IIRC, the client sends all moves to the server even for kibitzers.  The server just ignores moves by non-players.  It would just need to echo them back as suggestions and have the client manage drawing them to the board somehow.  I really like the idea of an arrow if we can get that working.

**Comment by arbrown** [comment]
So I implemented the basics of suggesting a move - spectators can try to make a move and the server will recognize that it is a suggestion and send an appropriate command back to the frontend.  That part was easy.  Then I spent 2-3 hours trying to overlay arrows in the right positions on the board.  It was surprisingly difficult.  I shelved that code for now and just made it send a chat message with the move notation instead for the meantime.  If anyone can come up with a good method of overlaying arrows that works with the janky React code, I'd love to hear it.

**Comment by jacoblangston** [comment]
I think a chat message is sufficient, although we may need to revamp the chat window.

**Comment by arbrown** [comment]
Yeah, it needs some CSS love, for sure, but I have no love for CSS.
